pkg/operation/botanist/nodelocaldns.go

```diff
@@ -59,5 +62,28 @@ func (b *Botanist) ReconcileNodeLocalDNS(ctx context.Context) error {
 	if b.Shoot.NodeLocalDNSEnabled {
 		return b.Shoot.Components.SystemComponents.NodeLocalDNS.Deploy(ctx)
 	}
+	if stillDesired, err := b.isNodeLocalDNSStillDesired(ctx); err != nil {
+		return err
+	} else if stillDesired {
+		// Leave NodeLocalDNS components in the cluster until all nodes have been rolled
+		return nil
+	}
```

Comment:
Nit: formatting/readability:
```suggestion

	if stillDesired, err := b.isNodeLocalDNSStillDesired(ctx); err != nil {
		return err
	} else if stillDesired {
		// Leave NodeLocalDNS components in the cluster until all nodes have been rolled
		return nil
	}

```

Reply:
Addressed with latests commit.